support-workers/src/main/scala/com/gu/support/workers/lambdas/CreateZuoraSubscription.scala

```diff
@@ -162,27 +170,51 @@ object ZuoraSubscriptionCreator {
     }
   }
 
+  // scalastyle:off cyclomatic.complexity
   def getEmailState(
     state: CreateZuoraSubscriptionState,
     accountNumber: ZuoraAccountNumber,
     subscriptionNumber: ZuoraSubscriptionNumber,
     paymentOrRedemptionData: Either[PaymentMethodWithSchedule, RedemptionData],
-    maybeGiftPurchase: Option[GiftPurchase],
-  ): SendThankYouEmailState =
+    maybeDigitalSubscriptionGiftCreationDetails: Option[DigitalSubscriptionGiftCreationDetails],
+  ): SendThankYouEmailState = {
+    val Purchase = Left
+    type Redemption = Right[PaymentMethodWithSchedule, RedemptionData]
     SendThankYouEmailState(
       requestId = state.requestId,
       user = state.user,
-      giftPurchase = maybeGiftPurchase,
-      product = state.product,
       analyticsInfo = state.analyticsInfo,
-      paymentOrRedemptionData = paymentOrRedemptionData,
-      firstDeliveryDate = state.firstDeliveryDate,
-      promoCode = state.promoCode,
+      productTypeCreated = paymentOrRedemptionData match {
+        case Purchase(PaymentMethodWithSchedule(paymentMethod, paymentSchedule)) =>
+          val purchaseInfo = PurchaseInfo(paymentMethod, paymentSchedule, state.promoCode, accountNumber.value, subscriptionNumber.value)
+          state.product match {
+            case product: Contribution =>
+              ContributionCreated(product, purchaseInfo)
+            case product: DigitalPack if product.readerType == ReaderType.Direct =>
+              DigitalSubscriptionDirectPurchaseCreated(product, purchaseInfo)
+            case product: DigitalPack if product.readerType == ReaderType.Gift =>
+              val giftPurchase = maybeDigitalSubscriptionGiftCreationDetails.get
+              DigitalSubscriptionGiftPurchaseCreated(product, giftPurchase.giftRecipient, giftPurchase.giftCode, giftPurchase.lastRedemptionDate, purchaseInfo)
+            case product: Paper =>
+              PaperCreated(product, purchaseInfo, state.firstDeliveryDate.get)
+            case product: GuardianWeekly =>
+              GuardianWeeklyCreated(product, state.giftRecipient.map(_.asWeekly.get), purchaseInfo, state.firstDeliveryDate.get)
+          }
+        case _: Redemption =>
+          state.product match {
+            case product: DigitalPack if product.readerType == ReaderType.Corporate =>
+              DigitalSubscriptionCorporateRedemptionCreated(product, subscriptionNumber.value)
+            case product: DigitalPack if product.readerType == ReaderType.Gift =>
+              DigitalSubscriptionGiftRedemptionCreated(product/*tbc*/)
+            case _ => throw new RuntimeException("could not create value state")
```

Comment:
this would be redeeming other products than digi, or other readertypes than corp/gift